frontend/src/app/warehouse-page/warehouse-page.component.ts

```diff
@@ -1,79 +1,83 @@
   styleUrl: './warehouse-page.component.css'
 })
 export class WarehousePageComponent implements OnInit, OnDestroy {
-  ordersPending = [
-    {selected: false},
-    {selected: false},
-    {selected: false},
-    {selected: false},
-    {selected: false},
-    {selected: false},
-    {selected: false},
-    {selected: false},
-    {selected: false},
-    {selected: false},
-    {selected: false},
-    {selected: false},
-    {selected: false},
-    {selected: false}
-  ];
-  ordersInProgress: {selected: boolean}[] = [];
+  pendingReservations$: Observable<Reservation[]>;
+  inProgressReservations: Reservation[] = [];
 
   constructor(
+    private warehouseService: WarehouseService,
     private router: Router,
-    private location: Location,
     private renderer: Renderer2,
     @Inject(DOCUMENT) private document: Document,
   ) {}
 
   ngOnInit(): void {
       this.renderer.setStyle(this.document.body, 'background-color', "#fff");
+      this.pendingReservations$ = this.warehouseService.pendingReservations$;
   }
 
   ngOnDestroy(): void {
     this.renderer.removeStyle(this.document.body, 'background-color');
   }
 
-  active(order: any) {
-    if (this.ordersPending.includes(order)) {
-      this.ordersPending.filter(el => el !== order).forEach(el => el.selected = false);
+  active(order: Reservation) {
+    if (this.inProgressReservations.includes(order)) {
+      this.inProgressReservations.filter(el => el !== order).forEach(el => el.selected = false);
     } else {
-      this.ordersInProgress.filter(el => el !== order).forEach(el => el.selected = false);
+      this.pendingReservations$.subscribe({
+        next: notifications => {
+          notifications.filter(el => el !== order).forEach(el => el.selected = false);
+        }
+      })
     }
-    
     order.selected = !order.selected;
   }
   
-  accept(order: any) {
-    const index = this.ordersPending.indexOf(order);
-    if (index > -1) {
-      this.ordersPending.splice(index, 1);
-      order.selected = false;
-      this.ordersInProgress.push(order);
-    }
+  accept(order: Reservation) {
+    this.pendingReservations$.subscribe({
+      next: notifications => {
+        const index = notifications.indexOf(order);
+        if (index > -1) {
+          notifications.splice(index, 1);
+          order.selected = false;
+          this.inProgressReservations.push(order);
+        }
+      }
+    });
   }
 
-  backToPending(order: any) {
-    const index = this.ordersInProgress.indexOf(order);
-    if (index > -1) {
-      this.ordersInProgress.splice(index, 1);
-      order.selected = false;
-      this.ordersPending.push(order);
-    }
+  backToPending(order: Reservation) {
+    this.pendingReservations$.subscribe({
+      next: notifications => {
+        const index = this.inProgressReservations.indexOf(order);
+        if (index > -1) {
+          this.inProgressReservations.splice(index, 1);
+          order.selected = false;
+          notifications.push(order);
```

Comment:
Reservations should be sorted by date in descending order after being moved back to the new section.